### DISCUSSION TITLE
Making a column type of 'enum'

### DISCUSSION THREAD

**Discussion by DangarStu** [saulpw/visidata#2096] [Q&A] [answered]
Hey there! I'm new to VisiData, but absolutely loving it, thank you so much for creating it, I've signed up to Patreon. :)

I've been looking through the docs and doing lots of searches but I can't see how to create a column where the value for a cell is selected from a constrained list of values. Is this possible?

Thanks in advance,
Stu

**Comment by saulpw** [comment, marked as answer]
Hey @DangarStu, welcome!  There's a `setcol-format-enum` which you can use to set mapping of values to more readable representation; for example if a field has `1` meaning "red" and `2` meaning "blue" you  use `setcol-format-enum` with `1=red 2=blue` and it would show the mapped value instead.

But otherwise no, enums don't really exist in VisiData.  What exactly would you want it to do?  It sounds like you're using VisiData for data entry; do you want to get an error if you try to input a bad value? or auto-complete when you enter the first letter of a value?

**Reply by DangarStu** [reply]
Thanks for the reply Saul, that sounds perfect for my needs. Thanks for all your work on VisiData!

**Reply by DangarStu** [reply]
Sorry, I didn't reply properly to your questions. Yes, in this case I am using VisiData for data entry and I just want to be able to enter quick values (such as integers) that represent long human-readable values. Auto/ tab complete is a great suggestion for a feature but it is no problem that it isn't currently supported.
